yt_dlp/extractor/youtube.py

```diff
@@ -496,16 +496,10 @@ def _initialize_consent(self):
         cookies = self._get_cookies('https://www.youtube.com/')
         if cookies.get('__Secure-3PSID'):
             return
-        consent_id = None
-        consent = cookies.get('CONSENT')
-        if consent:
-            if 'YES' in consent.value:
-                return
-            consent_id = self._search_regex(
-                r'PENDING\+(\d+)', consent.value, 'consent', default=None)
-        if not consent_id:
-            consent_id = random.randint(100, 999)
-        self._set_cookie('.youtube.com', 'CONSENT', 'YES+cb.20210328-17-p0.en+FX+%s' % consent_id)
+        socs = cookies.get('SOCS')
+        if socs and not socs.value.startswith('CAA'):  # not consented
+            return
+        self._set_cookie('.youtube.com', 'SOCS', 'CAI', secure=True)  # accept all (required for mixes)
```

Comment:
Why not just:
```suggestion
        if cookies.get('__Secure-3PSID') or cookies.get('SOCS'):
            return
        self._set_cookie('.youtube.com', 'SOCS', 'CAI', secure=True)  # accept all (required for mixes)
```

If user has exported consent cookie, we should always obey it, no?

Reply:
It was already doing that. The existence of SOCS does not mean the user has consented or not (in the issue it is set as soon as you go on YouTube).


SOCS stating with CAA means user has not consented (I.e. what YouTube sets by default above, neither rejected or accepted)